core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -232,6 +284,25 @@ sealed trait Message[+F[_]] extends Media[F] { self =>
 
   def mapK[F2[x] >: F[x], G[_]](f: F2 ~> G): SelfF[G] =
     self.change(entity = entity.translate(f))
+
+  /** Compiles the body stream to a single chunk and sets it as the
+    * body.  Replaces any `Transfer-Encoding: chunked` with a
+    * `Content-Length` header.  It is the caller's responsibility to
+    * assure there is enough memory to materialize the body.
+    */
+  def toStrict[F1[x] >: F[x]](implicit F: Concurrent[F1]): F1[SelfF[F1]] =
```

Comment:
This takes into account Ross's comment https://github.com/http4s/http4s/pull/6176#issuecomment-1075607483

Reply:
I think this could be `F[SelfF[Pure]]`.

Reply:
Yeah, I think so. Though this would mean we should add extra copying in cases of `Entity.{Empty|Strict}`, because using `covary` isn't an option (`[fs2.Pure] do not conform to method covary's type parameter bounds [F2[x] >: F[x]]`). But still, it's good to highlight the fact of effects evaluating.